lib/3scale/client/http_client.rb

```diff
@@ -65,14 +67,15 @@ def self.prepare
           require 'net/http/persistent'
         end
 
-        def initialize(host)
+        def initialize(host, port)
           super
           @http = ::Net::HTTP::Persistent.new
           @protocol = 'http'
         end
 
-        def ssl!
+        def ssl!(port)
           @protocol = 'https'
+          @port = port
```

Comment:
Maybe it would be better to set this in `initialize`? Then no need to pass `port` as parameter.